module/executiondatasync/optimistic_sync/execution_result/info_provider.go

```diff
@@ -94,6 +108,11 @@ func (e *Provider) ExecutionResultInfo(
 		return nil, fmt.Errorf("failed to choose execution nodes for block ID %v: %w", blockID, err)
 	}
 
+	// If criteria cannot be met, return an error
+	if len(subsetENs) < len(criteria.RequiredExecutors) {
+		return nil, optimistic_sync.NewCriteriaNotMetError(blockID)
+	}
+
```

Comment:
Similar to my previous comment, we don't need `criteria.RequiredExecutors` to match all ENs.
```suggestion
```